LibreNMS/Data/Source/SnmpResponse.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/*
+ * SnmpResponse.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2021 Tony Murray
+ * @author     Tony Murray <murraytony@gmail.com>
+ */
+
+namespace LibreNMS\Data\Source;
+
+use Illuminate\Support\Arr;
+use Illuminate\Support\Str;
+use Log;
+
+class SnmpResponse
+{
+    /**
+     * @var string
+     */
+    private $raw;
+    /**
+     * @var int
+     */
+    private $exitCode;
+
+    public function __construct(string $output, int $exitCode = 0)
+    {
+        $this->exitCode = $exitCode;
+        $this->raw = preg_replace('/Wrong Type \(should be .*\): /', '', $output);
+    }
+
+    public function isValid(): bool
+    {
+        $invalid = empty($this->raw)
+//            || $this->exitCode !== 0
+            || preg_match('/(No Such Instance|No Such Object|No more variables left|Authentication failure|Timeout: No Response from)/i', $this->raw, $matches);
```

Comment:
Very unlikely, but a response coooooould match this.
It's not printed on stderr or so instead?
Or maybe can restrict it to beginning of line or something?

Reply:
This is copied from the existing code.
I think we could remove the case insensitive switch though.